src/main/java/pl/gov/coi/pomocua/ads/BaseOffer.java

```diff
@@ -73,6 +77,16 @@ public void attachTo(User user) {
         this.userFirstName = user.firstName();
     }
 
+    @PostLoad
+    private void onLoad() {
+        Optional<PhoneDetails> phoneDetailsOpt = PhoneUtil.getPhoneDetails(phoneNumber);
+        if (phoneDetailsOpt.isPresent() && StringUtils.isEmpty(phoneCountryCode)) {
```

Comment:
Maybe we could check if `StringUtils.isEmpty(phoneCountryCode)` before parsing phone number with `PhoneUtil`?
This parsing could be a heavy operation which takes time, so it would be great to do it only if necessary.

Also, `Optional` has a `ifPresent()` method which could make this code a little bit prettier, but that's just my preference.

Reply:
good idea